Sources/PointFree/Account/Cancel.swift

```diff
@@ -0,0 +1,192 @@
+import Css
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+@testable import Tuple
+
+let confirmCancelResponse =
+  requireUser
+    <<< requireSubscription
+    <<< requireSubscriptionOwner
+    <<< requireStripeSubscription(^\.status != .canceled)
+    <| writeStatus(.ok)
+    >-> respond(confirmCancelView.contramap(lower))
+
+let cancelMiddleware =
+  requireUser
+    <<< requireSubscription
+    <<< requireSubscriptionOwner
+    <<< requireStripeSubscription(^\.status != .canceled)
+    <| map(lower)
+    >>> { conn -> IO<Conn<StatusLineOpen, Prelude.Unit>> in
+      let (subscription, data) = conn.data
+
+      return AppEnvironment.current.stripe.cancelSubscription(subscription.id)
+        .run
+        .map(^\.right)
+        .map(const(conn.map(const(unit))))
+    }
+    >-> redirect(to: .account)
+
+let reactivateMiddleware =
+  requireUser
+    <<< requireSubscription
+    <<< requireSubscriptionOwner
+    <<< requireStripeSubscription(^\.cancelAtPeriodEnd)
+    <| map(lower)
+    >>> { conn -> IO<Conn<StatusLineOpen, Prelude.Unit>> in
+      let (subscription, data) = conn.data
+
+      return AppEnvironment.current.stripe.reactivateSubscription(subscription)
```

Comment:
wanna add some `// TODO`s or create some Trello cards for all the emails we gotta send with these?